src/main/scala/dpla/ingestion3/OaiRecordHarvesterMain.scala

```diff
@@ -42,27 +43,27 @@ object OaiRecordHarvesterMain {
   val logger = LogManager.getLogger(OaiRecordHarvesterMain.getClass)
 
   def main(args: Array[String]): Unit = {
+    val oaiConf = new OaiHarvesterConf(args.toSeq)
 
-    validateArgs(args)
-    println(schemaStr)
+    // These are the minimum required parameters for an OAI harvest
+    val outputFile = oaiConf.outputDir.apply()
+    val endpoint = oaiConf.endpoint.apply()
+    val metadataPrefix = oaiConf.prefix.apply()
+    val provider = oaiConf.provider.apply()
 
-    val outputFile = args(0)
-    val endpoint = args(1)
-    val metadataPrefix = args(2)
-    val provider = args(3)
-
-    // This is an Option (as opposed to a String) b/c the param args(4) is optional.
-    val sets: Option[String] = if (args.isDefinedAt(4)) Some(args(4)) else None
-
-    // TODO rewrite this will better named parameters
-    val blacklistSets: Option[String] = if (args.isDefinedAt(5)) Some(args(5)) else None
+    // These are Options (as opposed to a String) b/c optional.
+    val sets: Option[String] = oaiConf.sets.toOption
+    val blacklistSets: Option[String] = oaiConf.blacklist.toOption
+    val sparkMaster: String = oaiConf.sparkMaster.apply()
 
     val verb = "ListRecords"
 
     Utils.deleteRecursively(new File(outputFile))
 
     // Initiate spark session.
     val sparkConf = new SparkConf().setAppName("Oai Record Harvest")
+    sparkConf.setMaster(sparkMaster)
```

Comment:
Looks like if this stays in, we're running this stuff through sbt rather than spark-submit. I like that idea for now, but not sure if it's workable when using EMR or not.